taxonomy/api.py

```diff
@@ -0,0 +1,245 @@
+"""APIs for lore taxonomy application"""
+
+from django.db import transaction
+
+from taxonomy.models import (
+    Vocabulary,
+    Term,
+)
+from learningresources.models import (
+    LearningResource,
+    LearningResourceType,
+    Repository,
+)
+
+
+# pylint: disable=no-member
+def get_vocabulary(vocabulary_id):
+    """
+    Lookup vocabulary given its id
+
+    Args:
+        vocabulary_id (int): Vocabulary id
+
+    Returns:
+        Vocabulary: The vocabulary from the database
+    """
+    return Vocabulary.objects.get(id=vocabulary_id)
```

Comment:
This could blow up if given a bad ID. A couple of options are [get_object_or_404](https://docs.djangoproject.com/en/1.8/topics/http/shortcuts/#get-object-or-404) or doing a `.filter()` and checking whether its `.count() == 0` and returning `None` if so. 

I like the 404 option better, because:

* You're not returning different data types (`None` versus a `Vocabulary` instance)
* If someone causes that then they did something wrong
* Calling code doesn't have to check if the returned values `is None` before trying to use it